analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CompareNaN.CSharp11.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+
+class CompareNaN
+{
+    void ListPattern()
+    {
+        object[] numbers = { 1, 2, double.NaN };
+
+        if (numbers is [1, 2, double.NaN]) // FN
+        {
+            Console.WriteLine("Test1");
+        }
+
+        if (numbers is [1, double.NaN, 3]) // FN
```

Comment:
Why do you have this second test case? Why is it different from the first one?

Reply:
The first one is comparing NaN to NaN, and the second Int32 to Nan, I thought that doing that should cover more expressively the pattern. Which one do you think should be removed?

Reply:
Discussed offline.